tests/geomultiscale-mapping/AxialGeoMultiscale.cpp

```diff
@@ -0,0 +1,88 @@
+#ifndef PRECICE_NO_MPI
+
+#include "testing/Testing.hpp"
+
+#include <precice/Participant.hpp>
+#include <vector>
+#include "testing/TestContext.hpp"
+
+using namespace precice;
+using precice::testing::TestContext;
+
+BOOST_AUTO_TEST_SUITE(Integration)
+BOOST_AUTO_TEST_SUITE(GeomultiscaleMapping)
+BOOST_AUTO_TEST_CASE(AxialGeoMultiscale)
+{
+  PRECICE_TEST("SolverOne"_on(1_rank), "SolverTwo"_on(1_rank))
+
+  using Eigen::Vector3d;
+
+  Participant cplInterface(context.name, context.config(), 0, 1);
+  if (context.isNamed("SolverOne")) {
+    auto     meshName = "MeshOne";
+    Vector3d posOne   = Vector3d::Constant(0.0);
+    auto     vid      = cplInterface.setMeshVertex(meshName, posOne);
+    auto     dataAID  = "DataOne";
+    auto     dataBID  = "DataTwo";
+
+    Vector3d valueDataB;
+
+    cplInterface.initialize();
+    double maxDt = cplInterface.getMaxTimeStepSize();
+    cplInterface.readData(meshName, dataBID, {&vid, 1}, maxDt, valueDataB);
+    Vector3d expected(0.0, 0.0, 8.0);
+    BOOST_TEST(valueDataB == expected);
+
+    while (cplInterface.isCouplingOngoing()) {
+      Vector3d valueDataA(0.0, 0.0, 8.0);
+      cplInterface.writeData(meshName, dataAID, {&vid, 1}, valueDataA);
+
+      cplInterface.advance(maxDt);
+      maxDt = cplInterface.getMaxTimeStepSize();
+
+      cplInterface.readData(meshName, dataBID, {&vid, 1}, maxDt, valueDataB);
+      expected << 0.0, 0.0, 8.0;
+      BOOST_TEST(valueDataB == expected);
```

Comment:
In contrast to the radial mapping, this seems to indeed be testing the feature:

1. `SolverOne`:
   - Defines a vector value at (0, 0, 0)
   - Reads `valueDataB` on that vertex, which is checked to be of value (0, 0, 8)
   - Writes `valueDataA`, which is of the same value ((0, 0, 8), maybe choosing different values here would prevent some accidents)
2. `SolverTwo`:
   - Defines one vertex at (0,0,0)
   - Writes `valueDataB`, which is vector of value (0, 0, 4)
   - Reads `valueDataA`, which is checked to be of value (0, 0, 16)

I have not checked the numbers, but where does the 2x multiplication factor come from?

Reply:
The 2x comes from the position of the vertex being the center of the flow profile, therefore the average value from the 1D participant is doubled to obtain the maximum velocity of the 3D flow profile.

Reply:
I see some assumptions that need to be documented in comments:

- SolverOne is the 1D code (we currently only have 3D vertices). This is not yet easy to guess (even when looking at the config).
- We are assuming a parabolic profile.

Reply:
On it!